### PR TITLE
Add PhpStorm

### PR DESCRIPTION
**Pull request by Lctrs**

### CODE REVIEW

**Comment by LeandroStanger** [comment]
bot, build com.jetbrains.PhpStorm

**Comment by LeandroStanger** [comment]
bot, build com.jetbrains.PhpStorm

File: com.jetbrains.PhpStorm.appdata.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    <screenshots>
+        <screenshot type="default">
+            <image>https://dashboard.snapcraft.io/site_media/appmedia/2019/08/welcome.png</image>
+        </screenshot>
+        <screenshot>
+            <image>https://dashboard.snapcraft.io/site_media/appmedia/2019/08/overview.png</image>
+        </screenshot>
+    </screenshots>
+
+    <categories>
```
Comment (bilelmoussaoui) [inline com.jetbrains.PhpStorm.appdata.xml]: Categories should be set in the desktop file instead.

**Comment by nedrichards** [comment]
bot, build com.jetbrains.PhpStorm

**Comment by x80486** [comment]
You might find this useful: https://gist.github.com/x80486/72e776d4847b7c2d1615e3b03bc6d25c for tracking updates. The `PhpStorm` code is `PS`.

File: com.jetbrains.PhpStorm.json [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+        "install -Dm0755 -t ${FLATPAK_DEST}/bin/ $(which git-upload-archive)",
+        "install -Dm0755 -t ${FLATPAK_DEST}/bin/ $(which git-upload-pack)"
+      ]
+    },
+    {
+      "name": "phpstorm",
+      "buildsystem": "simple",
+      "build-commands": [
+        "unzip phpstorm_logos.zip",
+        "install -Dm0755 apply_extra.sh ${FLATPAK_DEST}/bin/apply_extra",
```
Comment (bilelmoussaoui) [inline com.jetbrains.PhpStorm.json]: Put the apply_extra after installing the icon/desktop/metainfo files.

File: com.jetbrains.PhpStorm.json [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+        {
+          "type": "git",
+          "url": "https://gitlab.gnome.org/GNOME/libsecret.git",
+          "tag": "0.19.1",
+          "commit": "f576dfbe50f9aed22802ad0db0356d081f467d7a"
+        }
+      ]
+    },
+    {
+      "name": "git",
```
Comment (bilelmoussaoui) [inline com.jetbrains.PhpStorm.json]: Any reason of not just using the SDK as a runtime too? It's an IDE after all.

**Comment by bilelmoussaoui** [comment]
bot, build com.jetbrains.PhpStorm

**Comment by LeandroStanger** [comment]
bot, build com.jetbrains.PhpStorm

**Pull request closed without merging**